### DISCUSSION TITLE
Dependabot running when not enabled

### DISCUSSION THREAD

**Discussion by rcowsill** [community/community#22005] [New to GitHub]
<p>I just noticed a <a href="https://github.com/rcowsill/action-docker-layer-caching/pull/1" rel="noopener nofollow ugc">pull request from Dependabot</a> in one of my repositories. This is unexpected, as I haven’t enabled Dependency Graph or Dependabot in that repo:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/6/b6c56a6b55ff4e1edfea298c6b07588822f3ef9a.png" data-download-href="/uploads/short-url/q4RRmWSHTdKHtgiVNijuwAULqI2.png?dl=1" title="Settings" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/42620235/181068198-8bdf62c5-6e65-4921-bd6e-1f73c3fa079c.png" alt="Settings" data-base62-sha1="q4RRmWSHTdKHtgiVNijuwAULqI2" width="690" height="279" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/6/b6c56a6b55ff4e1edfea298c6b07588822f3ef9a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Settings</span><span class="informations">939×381 11.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This repo is a fork, and the owner of the source repo merges the Dependabot pull requests regularly.</p>
<p>Is this expected behaviour, and if so is there any way to prevent it?</p>

**Comment by airtower-luna** [comment]
<p>That is probably caused by a <code>.github/dependabot.yml</code> file in the repository, which configures Dependabot for version updates:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates" target="_blank" rel="noopener">Enabling and disabling version updates - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>From the description there you’d still need to enable version updates in a fork before that file is used. There’s no mention of how to disable it once enabled, but maybe that is a place to start looking? :sweat_smile:</p>

**Comment by rcowsill** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="151852">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p><a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates" class="inline-onebox" rel="noopener nofollow ugc">Enabling and disabling version updates - GitHub Docs</a></p>
</blockquote>
</aside>
<p>From that page I spotted this section about <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-and-disabling-version-updates#enabling-version-updates-on-forks" rel="noopener nofollow ugc">enabling Dependabot on forks</a>. That refers to the <code>Insights-&gt;Dependency Graph-&gt;Dependabot</code> page, which looks like this on my fork:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/5/152b5e3ec6d261bc8b0c5b42ba9e4a3f7745636c.png" data-download-href="/uploads/short-url/31gWrKnF4ArQCpoDbZ6lEdtyM8s.png?dl=1" title="Dependabot" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/42620235/181068202-d84b4573-58f1-426f-8267-b2e91efca1c2.png" alt="Dependabot" data-base62-sha1="31gWrKnF4ArQCpoDbZ6lEdtyM8s" width="690" height="145" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/5/152b5e3ec6d261bc8b0c5b42ba9e4a3f7745636c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Dependabot</span><span class="informations">943×199 4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I don’t remember enabling that, but maybe I clicked on it accidentally without noticing :sweat_smile:</p>
<p>The odd thing is that I still don’t have dependency graph enabled. Perhaps Dependabot is pulling the graph from upstream, but that seems like it’d break if I edit the dependencies in my fork :confused:</p>
<p>Unfortunately there doesn’t seem to be any way out of this situation. The documentation only gives one way to switch Dependabot off, and that’s to delete the dependabot.yml file. I don’t want to do that as it will make my fork’s default branch diverge from upstream. If anyone knows a way to fix for this I’d appreciate some pointers.</p>
<hr>
EDIT: I tried switching the three settings shown in my original post on/off to see if that would help. It doesn't seem to have changed anything, plus it won't let me switch off dependency graph:
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/1/f1c7a7b5ac0b572f6b4dd5482be57d6529bb90ca.png" data-download-href="/uploads/short-url/yuSPqyXrVULiApdrG7fSoRXlA3o.png?dl=1" title="deps" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/42620235/181068203-f7ac1c40-5d5d-440e-8034-822078d56738.png" alt="deps" data-base62-sha1="yuSPqyXrVULiApdrG7fSoRXlA3o" width="690" height="55" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/1/f1c7a7b5ac0b572f6b4dd5482be57d6529bb90ca_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">deps</span><span class="informations">922×74 2.62 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Notice how now it says public repos always have dependency graph enabled. I guess that’s how Dependabot was running even though I <em>appeared</em> to not have the dependency graph switched on.</p>

**Comment by mayakacz** [comment, marked as answer]
<p>This is a known issue, apologies for the confusion and your experience here.</p>
<p>Currently, <strong>when you fork a repo with Dependabot version updates enabled, your fork will also have Dependabot version updates enabled.</strong> This is because Dependabot version updates is configured with a dependabot.yml config file.</p>
<p>To disable Dependabot version updates, remove the dependabot.yml config file from your repo. (I realize this complicates suggesting changes, which is why this is a bug we are working to address today.)</p>
<blockquote>
<p>EDIT: I tried switching the three settings shown in my original post on/off to see if that would help.</p>
</blockquote>
<p>The settings on the Security tab only control Dependabot <em>security</em> updates, not also Dependabot version updates. Dependabot <em>version</em> updates are only controlled with the configuration file.</p>
<blockquote>
<p>Notice how now it says public repos always have dependency graph enabled. I guess that’s how Dependabot was running even though I appeared to not have the dependency graph switched on.</p>
</blockquote>
<p>Dependency Graph is on for all public repos and cannot be disabled on public repos. Dependabot <em>version</em> updates (which is what you are dealing with here) does not require the dependency graph - it doesn’t check your dependencies from the graph, only suggests edits to what’s in your lockfile.</p>

**Comment by sampaiodiego** [comment]
<p>are there any news on this regard?</p>

**Comment by thegcat** [comment]
<p>Is the proposed solution still to have a commit on master/main that contributors have to roll up each time they sync “their” master/main with that of the upstream project?</p>
